extensions/MikeDEV/peerjs.js

```diff
@@ -0,0 +1,929 @@
+          },
+          {
+            opcode: "hangupPeerCall",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate("hangup or decline call from peer [ID]"),
+            arguments: {
+              ID: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "B",
+              },
+            },
+          },
+          "---",
+          {
+            opcode: "whenPeerGetsMessage",
+            blockType: Scratch.BlockType.EVENT,
+            isEdgeActivated: false,
+            text: Scratch.translate(
+              "when I get a message from peer [ID] in channel [CHANNEL]"
+            ),
+            arguments: {
+              ID: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "B",
+              },
+              CHANNEL: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "default",
+              },
+            },
+          },
+
+          {
+            opcode: "readMessageFromPeer",
+            blockType: Scratch.BlockType.REPORTER,
+            text: Scratch.translate(
+              "message from peer [ID] in channel [CHANNEL]"
+            ),
+            arguments: {
+              ID: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "B",
+              },
+              CHANNEL: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "default",
+              },
+            },
+          },
+          {
+            opcode: "doesPeerHaveChannel",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: Scratch.translate("does [ID] have channel [CHANNEL]?"),
+            arguments: {
+              ID: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "B",
+              },
+              CHANNEL: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "blueberry",
+              },
+            },
+          },
+          {
+            opcode: "sendMessageToPeer",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate(
+              "send [MESSAGE] to peer [ID] using channel [CHANNEL]"
+            ),
+            arguments: {
+              MESSAGE: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "hello world",
+              },
+              ID: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "B",
+              },
+              CHANNEL: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "default",
+              },
+            },
+          },
+          {
+            opcode: "openNewPeerChannel",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate(
+              "open a new channel [CHANNEL] with peer [ID] and require messages to be ordered? [ORDERED]"
```

Comment:
Maybe this can be executed better? Think to the first review comment, but I'm not sure myself...